CMakeLists.txt

```diff
@@ -88,12 +88,11 @@ endif()
 
 # Example applications
 
-set(BUILD_EXAMPLES ON CACHE BOOL "Enable building the examples")
-
-if((CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME) AND BUILD_EXAMPLES)
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
```

Comment:
I removed `BUILD_EXAMPLES`, because the intention of was to offer an option to not build the examples when greentea-client is embedded in another project. But now we already have the check `if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)` which does it more elegantly.

Reply:
Just realised `BUILD_EXAMPLES` and `if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)` were introduced at the same time (#26), so I might have misunderstood the intention?

Reply:
The option is there so you can exclude building examples if you'd like to, even when not embedded in another project.

Reply:
But `-DBUILD_EXAMPLES=OFF` doesn't look simpler than passing `--target client`?

Reply:
`--target client` is an implementation detail, `BUILD_EXAMPLES` is part of a "public interface". My PR actually changes `BUILD_EXAMPLES` to be a CMake option, because that's a the semantically correct way of exposing options to the user, rather than using an internal cache variable.

Reply:
I'm going to revert this part to what it was before as it's not related to this PR. Will let your PR handle it or have discussion there.

Reply:
I'm still not sure whether keeping the option or using EXCLUDE_FROM_ALL and making the example targets part of the stable public interface is a better approach, though. 

Maybe it doesn't really matter too much which approach we take here, as the vast majority of users will be depending on this library in their test applications rather than building it alone just to run examples. So, maybe always building the examples if this is the current project is actually fine.

Reply:
Yea I agree it doesn't really matter here.

Bundling example(s) in a library's repository is quite common, so hopefully we'll reach a conclusion and follow that in the future. I've reverted this part for now.